### PR TITLE
Add object reduce

### PR DESCRIPTION
**Pull request by giuseppeg**
`reduce` but for objects.

N.B. for @angus-c 
I know that you favor loops over hof but it felt a bit overkill to re-write `Array.prototype.reduce`

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -98,6 +99,16 @@ filter({a1: 3, b1: 5, a2: 9}, (key, value) => key[0] == 'a'); // {a1: 3, a2: 9}
...
 ```
 
+### [just-reduce-object](https://www.npmjs.com/package/just-reduce)
+
+`npm install just-reduce-object`
+
+```js
+import reduce from 'just-reduce-object';
+
+reduce({a: 3, b: 5, c: 9}, (acc, key, value) => { acc[value]=key; return acc; }, {}); // {3: 'a', 5: 'b', 9: 'c'}
```
Comment (angus-c) [inline README.md]: line breaks

```
reduce({a: 3, b: 5, c: 9}, (acc, key, value) => { 
  acc[value]=key; 
  return acc; 
}, {}); // {3: 'a', 5: 'b', 9: 'c'}

File: packages/object-reduce/README.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
+## just-reduce-object
+
+Part of a [library](../../../../) of zero-dependency npm modules that do just do one thing.
+Guilt-free utilities for every occasion.
+
+Like `Array.prototype.reduce` but for `Object`s.
```
Comment (angus-c) [inline packages/object-reduce/README.md]: No need for this. Just usage exampe

File: packages/object-reduce/README.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+### Usage
```
Comment (angus-c) [inline packages/object-reduce/README.md]: Drop header

File: packages/object-reduce/README.md [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+  initiaValue
+);
+```
+
+* `obj` – source object
+* `predicate` – function to execute on each key-value pair in the object
+* `initialValue` – optional initial value for the accumulator
+
+### Examples
+
```
Comment (angus-c) [inline packages/object-reduce/README.md]: drop this - also no headers - just example

File: packages/object-reduce/README.md [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+}, {});
+```
+
+Without `reduce`:
+```js
+Object.keys(obj).reduce(function (accumulator, key, idx, origin) {
+  accumulator[key] = obj[key] * 2;
+  return accumulator;
+}, {});
+```
```
Comment (angus-c) [inline packages/object-reduce/README.md]: Just copy main README and no need for "without reduce" example
 
```js
reduce({a: 3, b: 5, c: 9}, (acc, key, value) => { 
  acc[value]=key; 
  return acc; 
}, {}); // {3: 'a', 5: 'b', 9: 'c'}
```

Then show another example with an initial accumualtor

File: packages/object-reduce/index.js [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+    // `previousValue` is the value associated to the first key
+    if (!hasInitialValue) {
+      previousValue = obj[array[0]];
+      hasInitialValue = true;
+    }
+    return predicate(previousValue, currentKey, obj[currentKey], currentIndex, array);
+  }
+
+  return Array.prototype.reduce.apply(Object.keys(obj), args);
+}
```
Comment (angus-c) [inline packages/object-reduce/index.js]: add closing newline

File: packages/object-reduce/package.json [hunk 7]
```diff
@@ -0,0 +1,21 @@
+{
+  "name": "just-reduce-object",
+  "version": "0.1.0",
```
Comment (angus-c) [inline packages/object-reduce/package.json]: 1.0.0

File: test/object-reduce/index.js [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+      }
+      return arguments[0];
+    };
+  }
+
+  // Without initialValue
+  var result1 = reduce(obj, getArgsInitialIndex(1));
+  t.ok(compare(result1, [3, 'b', 5, 1, expectedKeys]));
+
+  // With initiaValue
```
Comment (angus-c) [inline test/object-reduce/index.js]: initialValue

**Comment by angus-c** [comment]
Sorry - bunch of comments - mostly formatting.
Don't hate me but I want all READMEs to look consistent and be based only on usage examples

File: packages/object-reduce/README.md (hunk 5 above)
Comment (angus-c) [inline packages/object-reduce/README.md]: Make it look identical to main README section

Comment (giuseppeg) [inline packages/object-reduce/README.md]: Do you mean without an initial accumulator?

**Comment by giuseppeg** [comment]
@angus-c thanks for the code review.
Do you think that it is cool to initialise the accumulator with `obj[array[0]]` when `initialValue` is not provided/available? Would you rather set it to the `currentKey`?

**Comment by angus-c** [comment]
shuldn't it just be {} ?

**Pull request merged**